webapp/src/app/admin/contacts/contacts.component.ts

```diff
@@ -204,15 +204,23 @@ export class ContactsComponent implements OnInit {
                         throw new Error("cannot recover this user");
                     }
 
-                    progress(70, "Creating new device for recovery");
+                    progress(50, "Creating new device for recovery");
                     const now = new Date();
                     const name = sprintf("recovered by %s at %d/%02d/%02d %02d:%02d",
                         this.user.credStructBS.credPublic.alias.getValue(),
                         now.getFullYear(), now.getMonth() + 1, now.getDate(), now.getHours(), now.getMinutes());
                     const ephemeralIdentity = SignerEd25519.random();
+                    let newDarc: Darc;
```

Comment:
Hum, it's weird for me to mutate variable outside a closure, as it might get run latter, or multiple times, one doesn't know from a quick look.

Just an idea, see how it fits: `this.user.executeTransactions` can be updated to be
```ts
async executeTransactions<T>(
    addTxs: (tx: SpawnerTransactionBuilder) => Promise<T> | T,
    wait = 0
): Promise<T>
```
which allows to have `const newDarc = await this.user.executeTransactions` with a closure returning the darc.

This way, you avoid the closure potentially modify `newDarc` unexpectedly.

Reply:
I had the same thoughts, but does your proposition work? I thought I'd also have to change the `SpawnerTransactionBuilder`, so I didn't do it, rabbithole and everything.

Reply:
In fact it's enough - thanks, that's much cleaner.